solutions/cf-1627E.mdx

```diff
@@ -0,0 +1,138 @@
+---
+id: cf-1627E
+source: CF
+title: Not Escaping
+author: Chongtian Ma
+---
+
+[Official Analysis](https://codeforces.com/blog/entry/99067)
+
+## Alternate Explanation
+
+There can be at most $n$ * $m$ rooms, which is about $10^{10}$ total rooms and too much to process. However, not all of them are relevant (i.e. there are some rooms where it is not a start point or an end point for a ladder, so processing them would be extraneous). Therefore, coordinate compression can be used here.
+
+We can use shortest paths to solve the problem. Since we can move freely on a floor, we can perform dijkstra to find the minimal health to reach every room on that floor. Then, we can iterate over all the ladders on that floor and calculate the minimum health to travel to the endpoints of each ladder by using it.
+
+## Implementation
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+using ll = long long;
+
+struct ladder {
+	int from_row, from_col, to_row, to_col, health; 
+};
+
+int main() {
+	int t;
+	cin >> t;
+	
+	for (int test = 1; test <= t; test++) {
+		
+		int n, m, k;
+		cin >> n >> m >> k;
+		vector<ll> x(n + 1);
```

Comment:
at least use a better name for `x`

Reply:
the problem statement uses ```x``` bruh

Reply:
give a name, what does it mean in the context of the problem?

Reply:
it'll be confusing if i change the name, i just added a comment

Reply:
fair enough

Reply:
where is the comment o.O

Reply:
it magically disappeared i swear